src/core/settings.cpp

```diff
@@ -0,0 +1,79 @@
+#include "core/settings.h"
+
+Settings::Settings()
+{
+    Init();
+}
+
+void Settings::Init()
+{
+    Filesys& Fsys(Filesys::Get());
+    AutoFile F = Fsys.Open(boost::filesystem::path("settings.json"));
+    if (!F.get() || !F->IsValid())
+    {
+        return;
+    }
+    JsonReader Reader(*F);
+    if (!Reader.IsValid())
+    {
+        return;
+    }
+    mRoot = Reader.GetRoot();
+}
+
+std::string Settings::GetStr(std::string const& key, std::string const& default/*=""*/) const
```

Comment:
could just pass by value, like the other functions. the same amount of copies would happen